dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/RequestParamsValidator.java

```diff
@@ -0,0 +1,114 @@
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.tracker.export;
+
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.ACCESSIBLE;
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.CAPTURE;
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.CHILDREN;
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.DESCENDANTS;
+import static org.hisp.dhis.common.OrganisationUnitSelectionMode.SELECTED;
+
+import java.util.Set;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.hisp.dhis.common.OrganisationUnitSelectionMode;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.feedback.BadRequestException;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class RequestParamsValidator {
+
+  /**
+   * Validates that no org unit is present if the ou mode is ACCESSIBLE or CAPTURE. If it is, an
+   * exception will be thrown. If the org unit mode is not defined, SELECTED will be used by default
+   * if an org unit is present. Otherwise, ACCESSIBLE will be the default.
+   *
+   * @param orgUnits list of org units to be validated
+   * @param orgUnitMode
+   * @throws BadRequestException
+   */
+  public static OrganisationUnitSelectionMode validateOrgUnitMode(
+      Set<UID> orgUnits, OrganisationUnitSelectionMode orgUnitMode) throws BadRequestException {
+
+    if (orgUnitMode == null) {
+      orgUnitMode = orgUnits.isEmpty() ? ACCESSIBLE : SELECTED;
+    }
+
+    if (!orgUnits.isEmpty() && (orgUnitMode == ACCESSIBLE || orgUnitMode == CAPTURE)) {
+      throw new BadRequestException(
+          String.format(
+              "orgUnitMode %s cannot be used with orgUnits. Please remove the orgUnit parameter and try again.",
+              orgUnitMode));
+    }
+
+    if ((orgUnitMode == CHILDREN || orgUnitMode == SELECTED || orgUnitMode == DESCENDANTS)
+        && orgUnits.isEmpty()) {
+      throw new BadRequestException(
+          String.format(
+              "At least one org unit is required for orgUnitMode: %s. Please add an orgUnit or use a different orgUnitMode.",
+              orgUnitMode));
+    }
+
+    return orgUnitMode;
+  }
+
+  /**
+   * Validates that the org unit is not present if the ou mode is ACCESSIBLE or CAPTURE. If it is,
+   * an exception will be thrown. If the org unit mode is not defined, SELECTED will be used by
+   * default if an org unit is present. Otherwise, ACCESSIBLE will be the default.
+   *
+   * @param orgUnit the org unit to validate
+   * @param orgUnitMode
+   * @return
+   * @throws BadRequestException
+   */
+  public static OrganisationUnitSelectionMode validateOrgUnitMode(
+      UID orgUnit, OrganisationUnitSelectionMode orgUnitMode) throws BadRequestException {
```

Comment:
Can we reuse the above method by calling it from here with `validateOrgUnitMode(Set.of(orgUnit), mode)`?

Reply:
I thought about it, the thing is that orgUnit could be null at this point, so I'd need to prevent an NPE and call it with an empty set in that case. 
Instead of doing that, and given that this method will be deleted as soon as /events accept a list of org units, I decided to leave it like it is.